lib/bap_powerpc/bap_powerpc.mli

```diff
@@ -0,0 +1,614 @@
+
+(**
+
+   ## Intro
+
+    This module is the only one that needed to write lifter functions.
+    The main idea is to make a life of lifter writers as easy as
+    possible. This implies that only a few lines of code should be
+    enough to describe any instruction.
+
+    We introduce RTL - the language we expect to be very expressive,
+    with lots of details hidden under the hood, so a user should not
+    care about minor details.
+
+    So proposed usage is just to open it at the very beginning of your
+    module:
+
+    open Bap_powerpc.Std
+    ...
+
+    Note, that everywhere in this module, where some operation has
+    a notion of bit position (byte/word ...), it's assumed that
+    numeration starts from most significant bit (byte/word ...).
+
+
+   ## Lifter
+
+    Any lifter function must have two arguments.
+    The first one is a model of CPU that contains all information like
+    registers, memory access, instruction address, etc.
+    The second argument is an array of instruction operands, and its
+    user responsibility to treat each element of this array
+    according to instruction semantics. Those, for add instruction
+    which sum register and immediate and save a result in a register,
+    a user must point that the first and the second element of
+    operands array registers and the third argument is immediate.
+
+    In general, what one should do to add an instruction is to write
```

Comment:
It sounds to cryptic or, even, biblical. Also not sure that this is the right place for this sentence. Usually, one should start thy description, with such sentence (pun intended). Also try to follow the following simple rules when writing documentation:

1. Thouh shall not use the passive mode. 
2. Thouh shall not use complex sentences.

In the latter, I mean, that you should use too complex constructions, especially those that include lots of indirections and possessives such as "which", "those". In general, if your sentence has too much punctuation that simplify it. This is technical documentation, so there is no need to write nice prose.